adapters/ix/ix.go

```diff
@@ -1,267 +1,189 @@
 package ix
 
 import (
-	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/prebid/prebid-server/openrtb_ext"
-
-	"github.com/prebid/prebid-server/pbs"
-
-	"golang.org/x/net/context/ctxhttp"
-
+	"github.com/golang/glog"
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
 )
 
-// maximum number of bid requests
-const requestLimit = 20
+var maxRequests = 20 // it's not a const for the unit test convenience
 
 type IxAdapter struct {
 	http *adapters.HTTPAdapter
 	URI  string
 }
 
-// Name is used for cookies and such
-func (a *IxAdapter) Name() string {
-	return string(openrtb_ext.BidderIx)
-}
-
-func (a *IxAdapter) SkipNoCookies() bool {
-	return false
-}
-
-type indexParams struct {
-	SiteID string `json:"siteId"`
-}
-
-type ixBidResult struct {
-	Request      *callOneObject
-	StatusCode   int
-	ResponseBody string
-	Bid          *pbs.PBSBid
-	Error        error
-}
-
-type callOneObject struct {
-	requestJSON bytes.Buffer
-	width       uint64
-	height      uint64
-}
+func (a *IxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	logAsJSON("bid-request: %s", request)
```

Comment:
Adapters should not write to the log. It can overwhelm the host's logging backend.

Reply:
It'll write to the log only when started with "-v 3". I saw other adapters doing  that.

Reply:
I discussed this with the team. We don't want adapters to write the application log, as it could quickly escalate. The v level does help with that, but we don't see why it's necessary. PBS already has debug functionality to return this exact value when running an auction.

You're right. There are other adapters writing to the log. Most are during application startup and is resolved with https://github.com/prebid/prebid-server/pull/1532. The others though which we haven't noticed (Adhese, Gamoshi, Pubmatic, and Rubicon Legacy) and will work with them to remove those calls.